packages/serverless-component/serverless.js

```diff
@@ -237,22 +239,46 @@ class NextjsComponent extends Component {
 
     const defaultEdgeLambdaPublishOutputs = await defaultEdgeLambda.publishVersion();
 
-    const defaultCloudfrontInputs =
-      (inputs.cloudfront && inputs.cloudfront.defaults) || {};
+    let defaultCloudfrontInputs;
+    if (inputs.cloudfront && inputs.cloudfront.defaults) {
+      defaultCloudfrontInputs = inputs.cloudfront.defaults;
+      delete inputs.cloudfront.defaults;
+    } else {
+      defaultCloudfrontInputs = {};
+    }
+
+    // Add any custom cloudfront configuration
+    // this includes overrides for _next, static and api
+    Object.entries(cloudFrontConfigs).map(([path, config]) => {
+      cloudFrontOrigins[0].pathPatterns[path] = {
+        // spread the existing value if there is one
+        ...cloudFrontOrigins[0].pathPatterns[path],
+        // spread custom config
+        ...config,
+        "lambda@edge": {
+          ...(config["lambda@edge"] || {}),
+          "origin-request":
+            // dont set if the path is static or _next
+            // spread the supplied overrides
+            !["static/*", "_next/*"].includes(path) &&
+            `${defaultEdgeLambdaOutputs.arn}:${defaultEdgeLambdaPublishOutputs.version}`
+        }
+      };
+    });
+
     const cloudFrontOutputs = await cloudFront({
       defaults: {
         ttl: 0,
         allowedHttpMethods: ["HEAD", "GET"],
-        ...defaultCloudfrontInputs,
         forward: {
           cookies: "all",
           queryString: true,
-          cookies: "all",
-          queryString: true,
-          ...defaultCloudfrontInputs.forward
+          queryString: true
         },
+        ...defaultCloudfrontInputs,
```

Comment:
Could you make sure that `...defaultCloudfrontInputs` doesn't remove the `HEAD`, `GET` methods from the default cache behaviour.

Reply:
I've moved allowedHttpMethods to the end so it wont be overriden